controllers/registeredservice_controller.go

```diff
@@ -161,12 +167,182 @@ func (r *RegisteredServiceReconciler) addServiceToCatalog(ctx context.Context, s
 	return nil
 }
 
+func (r *RegisteredServiceReconciler) registerHealthcheck(ctx context.Context, rs *primazaiov1alpha1.RegisteredService) error {
+	// Make a CronJob that runs the healthcheck.  The CronJob handler will take
+	// care of the rest.
+	cronjob := batchv1.CronJob{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      rs.Name,
+			Namespace: rs.Namespace,
+		},
+	}
+	operation, err := controllerutil.CreateOrUpdate(ctx, r.Client, &cronjob, func() error {
+		var one int32 = 1
+		var two int32 = 2
+		var thirty int64 = 30
+		var uid int64 = 65530
+		t := true
+		f := false
+		container := corev1.Container{
+			Name:            "healthcheck",
+			Command:         rs.Spec.HealthCheck.Container.Command,
+			Image:           rs.Spec.HealthCheck.Container.Image,
+			ImagePullPolicy: corev1.PullIfNotPresent,
+			SecurityContext: &corev1.SecurityContext{
+				Capabilities: &corev1.Capabilities{
+					Drop: []corev1.Capability{"ALL"},
+				},
+				Privileged:               &f,
+				RunAsNonRoot:             &t,
+				AllowPrivilegeEscalation: &f,
+			},
+		}
+		cronjob.Spec = batchv1.CronJobSpec{
+			// run every n minutes
+			Schedule:                   fmt.Sprintf("*/%d * * * *", rs.Spec.HealthCheck.Container.Minutes),
+			ConcurrencyPolicy:          batchv1.ForbidConcurrent,
+			SuccessfulJobsHistoryLimit: &one,
+			FailedJobsHistoryLimit:     &one,
+			JobTemplate: batchv1.JobTemplateSpec{
+				Spec: batchv1.JobSpec{
+					ActiveDeadlineSeconds: &thirty,
+					BackoffLimit:          &two,
+					Parallelism:           &one,
+					Template: corev1.PodTemplateSpec{
+						Spec: corev1.PodSpec{
+							Containers:    []corev1.Container{container},
+							RestartPolicy: corev1.RestartPolicyNever,
+							SecurityContext: &corev1.PodSecurityContext{
+								RunAsUser:    &uid,
+								RunAsGroup:   &uid,
+								RunAsNonRoot: &t,
+								SeccompProfile: &corev1.SeccompProfile{
+									Type: corev1.SeccompProfileTypeRuntimeDefault,
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+		return ctrl.SetControllerReference(rs, &cronjob, r.Scheme)
+	})
+
+	if err != nil {
+		return err
+	}
+
+	l := log.FromContext(ctx).WithValues("cronjob", cronjob.Name, "namespace", cronjob.Namespace)
+	if operation == controllerutil.OperationResultCreated {
+		l.Info("Created healthcheck cronjob")
+	} else {
+		l.Info("Modified healthcheck cronjob", "operation", operation)
+	}
+
+	return nil
+}
+
+func (r *RegisteredServiceReconciler) handleHealthcheck(ctx context.Context, rs *primazaiov1alpha1.RegisteredService) error {
+	l := log.FromContext(ctx).WithValues("namespace", rs.Namespace, "registered service", rs.Name)
+	cronjob := batchv1.CronJob{}
+	err := r.Get(ctx, types.NamespacedName{Name: rs.Name, Namespace: rs.Namespace}, &cronjob)
+	if k8errors.IsNotFound(err) {
+		l.Info("creating registered service healthcheck")
+		err := r.registerHealthcheck(ctx, rs)
+		rs.Status.State = primazaiov1alpha1.RegisteredServiceStateUnknown
+		return err
+	}
```

Comment:
what if it is another error? if it's a network error it may still be that the CronJob is not existing, also `cronjob` will be `nil` in the following